Sources/Modules/Yggdrasil/Furnace/Items/Accessories/HotAirBalloon.cs

```diff
@@ -43,9 +43,21 @@ public override void PreUpdate()
 	{
 		if (HotAirBalloonEnable)
 		{
-			if (Main.rand.NextBool(6))
+			if (!HotAirBalloonEnable)
 			{
-				Dust.NewDust(Player.position, Player.width, Player.height, DustID.Firefly);
+				return;
+			}
+
+			// Check if the player is moving
+			if (Player.velocity == Vector2.Zero)
```

Comment:
Have you ever considered a player in the air?
Terraria treat player in the air by giving them a slight vertical speed (0.00001)

Reply:
fixed